runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -120,7 +122,8 @@ end
 ---@private
 local function diagnostic_vim_to_lsp(diagnostics)
   return vim.tbl_map(function(diagnostic)
-    return vim.tbl_extend("error", {
+    return vim.tbl_extend("keep", {
+      -- "keep" the below fields over any duplicate fields in diagnostic.user_data.lsp
```

Comment:
This function looks like it's only called by deprecated `vim.lsp.diagnostic` functions. Should I just not change this function if that's the case?